python/wikibot/en_wp_dates.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Class for extracting birth death datesfrom English Wikipedia articles"""
+
+import sling
+import sling.flags as flags
+import re
+
+class ExtractWikipediaDates:
+  def __init__(self):
+    self.kb = sling.Store()
+    self.kb.lockgc()
+    print "loading kb"
+    self.kb.load("local/data/e/wiki/kb.sling", snapshot=True)
+    print "kb loaded"
+    self.instanceof = self.kb['P31']
+    self.date_of_birth = self.kb['P569']
+    self.date_of_death = self.kb['P570']
+    self.human = self.kb['Q5']
+    self.item = self.kb["item"]
+    self.facts = self.kb["facts"]
+    self.provenance = self.kb["provenance"]
+    self.url = self.kb["url"]
+    self.method = self.kb["method"]
+
+    self.kb.freeze()
+
+    self.months = {
+      "January": 1,
+      "February": 2,
+      "March": 3,
+      "April": 4,
+      "May": 5,
+      "June": 6,
+      "July": 7,
+      "August": 8,
+      "September": 9,
+      "October": 10,
+      "November": 11,
+      "December": 12
+    }
+
+  """Sling Date from regex match"""
+  def date_from_match(self, offset, match):
+    if match.group(1 + offset):
+      # DD MONTH YYYY format
+      date = sling.Date(int(match.group(3 + offset)),
+                        self.months[match.group(2 + offset)],
+                        int(match.group(1 + offset)))
+    else:
+      # MONTH DD, YYYY format"""
+      date = sling.Date(int(match.group(6 + offset)),
+                        self.months[match.group(4 + offset)],
+                        int(match.group(5 + offset)))
+    return date
+
+  def run(self):
+    month = "(January|February|March|April|May|June|July|August|September|"
```

Comment:
What about "(" + self.months.values().join("|") + ")"

Reply:
Yeah -- works if you do "|".join(months.keys()) instead. Fixed.